site/content/extend/plugins/webapp/interaction-with-redux.md

```diff
@@ -0,0 +1,502 @@
+    return state.entities.users.profiles[getCurrentUserId(state)];
+}
+```
+
+**getUsers**
+
+Retrieves all user profiles from the Redux store
+
+```javascript
+export function getUsers(state: GlobalState): IDMappedObjects<UserProfile> {
+    return state.entities.users.profiles;
+}
+```
+
+**getChannel**
+
+Retrieves a channel as it exists in the store without filling in any additional details such as the display_name for DM/GM channels.
+
+```javascript
+export function getChannel(state: GlobalState, id: string) {
+    return getAllChannels(state)[id];
+}
+```
+
+**getCurrentChannelId**
+
+Retrieves the channel ID of the current channel from the Redux store
+
+```javascript
+export function getCurrentChannelId(state: GlobalState): string {
+    return state.entities.channels.currentChannelId;
+}
+```
+
+**getCurrentChannel**
+
+Retrives the complete channel info of the current channel from the Redux Store
+
+```javascript
+export const getCurrentChannel: (state: GlobalState) => Channel = createSelector(
+    getAllChannels,
+    getCurrentChannelId,
+    (state: GlobalState): UsersState => state.entities.users,
+    getTeammateNameDisplaySetting,
+    (allChannels: IDMappedObjects<Channel>, currentChannelId: string, users: UsersState, teammateNameDisplay: string): Channel => {
+        const channel = allChannels[currentChannelId];
+
+        if (channel) {
+            return completeDirectChannelInfo(users, teammateNameDisplay, channel);
+        }
+
+        return channel;
+    },
+);
+```
+
+**getPost**
+
+Retrieves the specific post associated with the supplied `postID` from the Redux Store
+
+```javascript
+export function getPost(state: GlobalState, postId: $ID<Post>): Post {
+    return getAllPosts(state)[postId];
+}
+```
+
+**getCurrentTeamId**
+
+Retrieves the Team ID of the current team from the Redux store
+
+```javascript
+export function getCurrentTeamId(state: GlobalState) {
+    return state.entities.teams.currentTeamId;
+}
+```
+
+**getCurrentTeam**
+
+Retrieves the team info of the current team from the Redux store
+
+```javascript
+export const getCurrentTeam: (state: GlobalState) => Team = createSelector(
+    getTeams,
+    getCurrentTeamId,
+    (teams, currentTeamId) => {
+        return teams[currentTeamId];
+    },
+);
+```
+
+**getCustomEmojisByName**
+
+Retrieves the the specific emoji associated with the supplied `customEmojiName` from the Redux Store
+
+```javascript
+export const getCustomEmojisByName: (state: GlobalState) => Map<string, CustomEmoji> = createSelector(
+    getCustomEmojis,
+    (emojis: IDMappedObjects<CustomEmoji>): Map<string, CustomEmoji> => {
+        const map: Map<string, CustomEmoji> = new Map();
+
+        Object.keys(emojis).forEach((key: string) => {
+            map.set(emojis[key].name, emojis[key]);
+        });
+
+        return map;
+    },
+);
+```
+
+**getConfig**
+
+Retrieves the config of the server for admin analysis
```

Comment:
```suggestion
Retrieves the config of the server for System Admin analysis.
```

Reply:
This section has been removed as suggested by @larkox .